pkg/agent/testdata/RawUbuntuContainerd/line70.sh

```diff
@@ -388,6 +388,23 @@ ensureDHCPv6() {
 ensureKubelet() {
     KUBELET_DEFAULT_FILE=/etc/default/kubelet
     mkdir -p /etc/default
+
+    if semverCompare ${KUBERNETES_VERSION:-"0.0.0"} "1.29.0"; then
+        imdsOutput=$(curl -s -H Metadata:true --noproxy "*" --max-time 5 "http://169.254.169.254/metadata/instance?api-version=2021-02-01" 2> /dev/null)
```

Comment:
any risk that this will diverge from what cloud-node-manager sets?

Before k8s 1.29 kubelet would try to infer the IP address from Linux routes and interfaces, but would set only the IPv4 address, so there was already some difference between kubelet and cloud-node-manager.

Reply:
if cloud-manager sets the IP differently after, is this a bad thing ? could the initial `--node-ip` be temporary and only required for quicker bootstrapping ?

Reply:
right, I think that's the behavior this PR is trying to restore for us specifically, I guess upstream decided to stop doing that since hostNetwork pods created by cloud-provider assigned an external IP would keep the original IP used at kubelet start-time

that's at least my understanding from reading the linked issue/PR

Reply:
`but would set only the IPv4 address` I see that we'd now be setting both v4 and v6 addresses, is that potentially an issue?

Reply:
I think setting both IPv4 and IPv6 is an improvement over what we had before. Previously there was a race where kubelet would set IPv4 only, then cloud-node-manager would set IPv4+IPv6 later. This also matches the behavior on Windows,  see https://github.com/Azure/AgentBaker/pull/4148.

I'd be open to setting just IPv4 if we think that's less risky, but I'd prefer to align with what cloud-node-manager and Windows CSE do.

Reply:
updated to set just ipv4 address, since that's the only one we can get from nmagent

Reply:
changed it back to setting both ipv4 and ipv6, to match what cloud-node-manager does. This also solves the race condition where host network pods are assigned only IPv4 if they schedule before cloud-node-manager updates the IP.